query-request.hh

```diff
@@ -366,4 +368,28 @@ public:
     friend std::ostream& operator<<(std::ostream& out, const read_command& r);
 };
 
+struct forward_request {
+    enum class reduction_type {
+        count,
+    };
+    reduction_type type;
+    std::vector<sstring> selected_columns;
+
+    query::read_command cmd;
+    dht::partition_range_vector pr;
+
+    db::consistency_level cl;
+    lowres_clock::time_point timeout;
+    int32_t page_size;
+    bool skip_metadata;
+
+    friend std::ostream& operator<<(std::ostream& out, const forward_request& r);
+    friend std::ostream& operator<<(std::ostream& out, const reduction_type& r);
+};
+
+struct forward_result {
+    bytes_opt query_response;
```

Comment:
Some words about how this is serialized.

I think we need to support multiple results, in case we have `SELECT min(x), max(x), avg(x) FROM tab`, We don't need to support it on the first iteration, but let's have the protocol prepared. I guess that means std::vector<reduction_type> too (and we'll throw if v.size() != 1).